### DISCUSSION TITLE
Default to PR titles for squash merge commit messages REST API

### DISCUSSION THREAD

**Discussion by electriquo** [community/community#24592] [Apps, API and Webhooks]
<p>Following <a href="https://github.blog/changelog/2022-05-11-default-to-pr-titles-for-squash-merge-commit-messages/" rel="noopener nofollow ugc">Default to PR titles for squash merge commit messages</a>, is there any REST API to manage the option (in the same manner that <a href="https://docs.github.com/en/rest/repos/repos" rel="noopener nofollow ugc"> <code>allow_squash_merge</code></a> can be managed with REST API)?</p>

**Comment by cfarrend** [comment]
<p>Would be great to have this feature. I was surprised when I didn’t see an API option for it.</p>

**Comment by electriquo** [comment]
<p>i guess it is not going to be implemented, like many other missing REST API,<br>
e.g. <a href="https://github.community/t/require-deployments-to-succeed-before-merging-rest-api/244642" class="inline-onebox">Require deployments to succeed before merging REST API</a></p>

**Comment by freef4ll** [comment]
<p>Yes, it would be great to have, the <code>squash-pr-title</code> feature is great and it looks like the API for this was missed <a href="https://github.blog/changelog/2022-05-11-default-to-pr-titles-for-squash-merge-commit-messages" rel="noopener nofollow ugc">when it was introduced</a></p>
<pre><code class="lang-auto">gh api -H "Accept: application/vnd.github.v3+json" /repos/my_org/my_repo | jq | grep _merge
  "allow_squash_merge": true,
  "allow_merge_commit": false,
  "allow_rebase_merge": true,
  "allow_auto_merge": false,
  "delete_branch_on_merge": false,
</code></pre>
<p>The use case would be to have <a href="https://github.com/github/safe-settings" class="inline-onebox" rel="noopener nofollow ugc">GitHub - github/safe-settings</a> to control what options are set across all repos.</p>

**Comment by electriquo** [comment, marked as answer]
<p><a href="https://docs.github.com/en/rest/repos/repos#create-an-organization-repository" rel="noopener nofollow ugc"><code>use_squash_pr_title_as_default</code> was introduced</a></p>
